### PR TITLE
Use buffer module which support node.js and browser

### PR DESCRIPTION
**Pull request by xxkl1**
The Wepack5 haven't  provided polyfills for node.js core modules, and Vite does't suppot polyfill for buffer.

It will occur error on browser using the Wepack5 and Vite.

Because of the Browserify have used buffer polyfill, it is fine on browser.

I would suggest using buffer polyfill by add npm module buffer like Browserify.

link: https://github.com/mcollina/hyperid/issues/9#issuecomment-1010367066

reproduction:
Webpack5: https://stackblitz.com/edit/node-yctv93?file=src%2Findex.js,package.json
Vite: https://stackblitz.com/edit/node-bedrwr?file=package.json

### CODE REVIEW

File: hyperid.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 const uuidv4 = require('./uuid')
 const parser = require('uuid-parse')
 const maxInt = Math.pow(2, 31) - 1
-const Buffer = require('buffer').Buffer
+const Buffer = require('buffer/').Buffer
```
Comment (mcollina) [inline hyperid.js:6]:
```suggestion
const Buffer = require('buffer/').Buffer
```

Please use the global `Buffer` object if it is available

Comment (xxkl1) [reply hyperid.js]:
```javascript
function loadBuffer () {
  const b = require('buffer')
  // use third party module if no Buffer module
  return b
    ? b.Buffer
    : require('buffer/').Buffer
}
```
I have added a function to load Buffer, please review my new commit if have time.

**Comment by DanielRuf** [comment]
Why was this closed?

**Comment by mcollina** [comment]
thiw now conflicts

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
This should really include a test

**Comment by xxkl1** [comment]
> This should really include a test

Let me think about how to add test from this case.

**Comment by mcollina** [comment]
closing for no activity

**Pull request closed without merging**